src/timeline/app.js

```diff
@@ -51,32 +64,29 @@ $(document).ready(function () {
 
     track_controls.height(new_track_height);
     $("#scrolling_tracks").height(new_track_height);
-    body_object.scope().playhead_height = $("#track-container").height();
-    $(".playhead-line").height(body_object.scope().playhead_height);
+      body_object.scope().playhead_height = $("#track-container").height();
+      $(".playhead-line").height(body_object.scope().playhead_height);
   });
 
-  // Check for Qt Integration
-  if (typeof timeline !== "undefined") {
-    timeline.qt_log("Qt Found!");
-    body_object.scope().enableQt();
-    timeline.page_ready();
-    body_object.scope().setThumbAddress(timeline.get_thumb_address());
-  }
-
-  // Manually trigger the window resize code (to verify it runs at least once)
-  $(window).trigger("resize");
-
   // Bind to keydown event (to detect SHIFT)
   body_object.keydown(function (event) {
     if (event.which === 16) {
+      if (timeline) {
+        timeline.qt_log("Shift pressed!");
+      }
       body_object.scope().shift_pressed = true;
     }
   });
```

Comment:
So, funny story about these. Turns out, [`scope()` is a debugging tool](https://docs.angularjs.org/guide/production#disabling-debug-data), and we'd actually stand to make a significant performance gain if we could turn _off_ Angular debugging on the Timeline. But then `scope()` wouldn't work from JQuery.

Unfortunately, I have tried _desperately_ to get shift-tracking working with AngularJS's own `ng-keydown` and `ng-keyup`, and I have never gotten it to work anything close to reliably.

Reply:
Yeah, if we could disable debugging, but still somehow access Angular, that would work fine. But I have no idea how to do that. If I have time, I'll do some research on this.

Reply:
I was thinking, in _theory_ we could add hidden `<input>`s to the HTML, and store the data involved in **those**. Then, JQuery could update the values without needing `scope()`, and Angular could access them directly... well, however we want, really. Using jQuery selectors, using `angular.element()`, etc.

I'm thinking something like,
```html
<html><body><stuff stuff stuff>
<input type="hidden" id="shift_state" value="0">
<script>
    $("body").keydown(function (event) {
        if (event.which ===  16) $("#shift_state").val(1);
    });
    $("body").keyup(function (event) {
        if (event.which ===  16) $("#shift_state").val(0);
    });
</script>
<script>
// currently, the only place where shift_pressed is ever read...
function moveBoundingBox(scope, previous_x, previous_y, x_offset, y_offset, left, top) {
    var snapping_result = Object();
    snapping_result.left = left;
    snapping_result.top = top;

    // Check for shift key
    if (parseInt($("#shift_state").val(), 2) === 1) {
        // freeze X movement
        x_offset = 0;
    }
</script>
```
I mean, I can tell you that with the timeline loaded in a browser window and the debugging console open, I can _see_ the `value=""` attribute of the `<input>` flipping from `"0"` to `"1"` to `"0"` as I press and release <kbd>Shift</kbd>, so it at least works to that extent...

Reply:
@jonoomph 
> Yeah, if we could disable debugging, but still somehow access Angular, that would work fine. But I have no idea how to do that. If I have time, I'll do some research on this.

**\@me**
> I was thinking, in _theory_ we could add hidden `<input>`s to the HTML, and store the data involved in **those**. Then, JQuery could update the values without needing `scope()`, and Angular could access them directly... well, however we want, really. Using jQuery selectors, using `angular.element()`, etc.

...Actually I suspect there's no point in doing that, and I doubt we'll be able to get away from `scope()` any time soon. I didn't realize until yesterday that the _entire_ Qt < == > JS interface depends on `scope()`:
https://github.com/OpenShot/openshot-qt/blob/0058f87dbdd5c561c1247ab0db0edf590be19516/src/windows/views/timeline_webview.py#L55-L56

Maybe if we were to rewrite their entire communication layer as a WebChannel, or something like that, then we could get away from `scope()`. But for now, it seems it's here to stay.